lib/xapi-stdext-date/date.ml

```diff
@@ -100,7 +90,4 @@ let assert_utc _ = ()
 
 let never = of_float 0.0
 
-let eq x y = match x, y with
```

Comment:
Ah I see there could've been a case where Legacy and UTC represented the same time, but equal wouldn't compare them. Perhaps why the unit test failed in xapi. Does it pass now?

Reply:
i believe the problem was caused by the fact that previously `"2020-11-11T11:11:11Z" |> of_string` was not the same as `"2020-11-11T11:11:11Z" |> of_string |> to_string |> of_string` ... the first would have been a `UTC ptime` and the other a `Legacy string`

representing everything as a ptime fixes that - the tests pass now